cabal-install/Distribution/Client/Dependency/Modular/Builder.hs

```diff
@@ -125,8 +125,13 @@ build = ana go
     go    (BS { index = _  , next = OneGoal (OpenGoal (Simple (Pkg _ _          ) _) _ ) }) =
       error "Distribution.Client.Dependency.Modular.Builder: build.go called with Pkg goal"
     go bs@(BS { index = idx, next = OneGoal (OpenGoal (Simple (Dep qpn@(Q _ pn) _) _) gr) }) =
+      -- If the package does not exist in the index, we construct an emty PChoiceF node for it
+      -- After all, we have no choices here. Alternatively, we could immediately construct
+      -- a Fail node here, but that would complicate the construction of conflict sets.
+      -- We will probably want to give this case special treatment when generating error
+      -- messages though.
       case M.lookup pn idx of
-        Nothing  -> FailF (toConflictSet (Goal (P qpn) gr)) (BuildFailureNotInIndex pn)
+        Nothing  -> PChoiceF qpn gr (P.fromList [])
```

Comment:
@kosmikus Maybe I misunderstood, but I thought part of the change to use an empty `PChoice` involved adding the current variable to the initial conflict set that is given to the backjump function.  Then `initial` would contain the current variable and the goal reason variable.  I meant to say that I thought using both variables was more correct than using just the goal reason, regardless of how we deal with unknown packages.

I think it makes sense to revert the rest of the empty `POption` change, because it complicates the log processing.